cmd/govim/vimstate.go

```diff
@@ -113,7 +113,18 @@ func (v *vimstate) setConfig(args ...json.RawMessage) (interface{}, error) {
 	if !vimconfig.EqualBool(v.config.HighlightDiagnostics, preConfig.HighlightDiagnostics) {
 		if v.config.HighlightDiagnostics == nil || !*v.config.HighlightDiagnostics {
 			// HighlightDiagnostics is now not on - remove existing text properties
-			v.redefineHighlights([]types.Diagnostic{}, true)
+			v.BatchStart()
+			for bufnr, buf := range v.buffers {
```

Comment:
It also bothers me that we leak this sort of detail into the handling of config changes. Worth fixing now, or would you like to leave that for a later PR?

Reply:
That's a good point. I did consider breaking out the feature to a `removeHighlights` but went with this since it is how it's done for both signs and quickfix entries above.